presto-main/src/test/java/com/facebook/presto/type/TestDecimalOperators.java

```diff
@@ -72,7 +72,7 @@ public void testAdd()
         assertInvalidFunction("DECIMAL '-99999999999999999999999999999999999999' + DECIMAL '-99999999999999999999999999999999999999'", "DECIMAL result exceeds 38 digits: -199999999999999999999999999999999999998");
 
         // test null
-        assertFunction("NULL + DECIMAL '-2'", createDecimalType(1, 0), null);
+        assertFunction("NULL + DECIMAL '-2'", createDecimalType(2, 0), null);
```

Comment:
That is the problem. In deed arithmetic operators with `NULL` should have just other semantic. They should produce the same type as the first argument, without applying any type calculation rules from the actual arithmetic operators. But i checked how it works in `MySQL` - they don't care, and they  just return what they want. So, probably that is fine. Leave it for now.